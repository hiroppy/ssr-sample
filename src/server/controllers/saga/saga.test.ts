import httpMocks, { MockResponse } from 'node-mocks-http';
import { getAll, post } from '.';

test('should return the getAll of api response', () => {
  const req = httpMocks.createRequest({
    query: {
      maxLength: 1
    }
  });
  const res = httpMocks.createResponse();
  const data = getAll(req, res) as MockResponse<ReturnType<typeof getAll>>;

  expect(data._getJSONData()).toMatchSnapshot();
});

test('should return the post of api response', () => {
  const res = httpMocks.createResponse();

  {
    const req = httpMocks.createRequest({
      params: {
        id: 1
      }
    });
    const data = post(req, res) as MockResponse<ReturnType<typeof post>>;

    expect(data._getJSONData()).toMatchSnapshot();
  }

  {
    const req = httpMocks.createRequest({
      params: {
        id: 1000
      }
    });
    const data = post(req, res) as MockResponse<ReturnType<typeof post>>;

    expect(data._getJSONData()).toMatchSnapshot();
  }
});
