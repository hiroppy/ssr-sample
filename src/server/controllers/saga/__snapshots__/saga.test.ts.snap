// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return the getAll of api response 1`] = `
Object {
  "body": Array [
    Object {
      "code": "test-code",
      "description": "aaaa",
      "id": 1,
      "likeCount": 0,
      "name": "pages.tsx",
    },
  ],
}
`;

exports[`should return the post of api response 1`] = `
Object {
  "body": Object {
    "code": "test-code",
    "description": "aaaa",
    "id": 1,
    "likeCount": 1,
    "name": "pages.tsx",
  },
}
`;

exports[`should return the post of api response 2`] = `
Object {
  "body": Object {
    "code": "test-code",
    "description": "aaaa",
    "id": 1,
    "likeCount": 1,
    "name": "pages.tsx",
  },
}
`;
